src/config.rs

```diff
@@ -0,0 +1,335 @@
+
+    pub fn should_ignore(&self, ty: &RequestType) -> bool {
+        match ty {
+            RequestType::GETPWBYNAME => self.ignore_getpwbyname,
+            RequestType::GETPWBYUID => self.ignore_getpwbyuid,
+            RequestType::GETGRBYNAME => self.ignore_getgrbyname,
+            RequestType::GETGRBYGID => self.ignore_getgrbygid,
+            RequestType::GETHOSTBYNAME => self.ignore_gethostbyname,
+            RequestType::GETHOSTBYNAMEv6 => self.ignore_gethostbynamev6,
+            RequestType::GETHOSTBYADDR => self.ignore_gethostbyaddr,
+            RequestType::GETHOSTBYADDRv6 => self.ignore_gethostbyaddrv6,
+            RequestType::SHUTDOWN => self.ignore_shutdown,
+            RequestType::GETSTAT => self.ignore_getstat,
+            RequestType::INVALIDATE => self.ignore_invalidate,
+            RequestType::GETFDPW => self.ignore_getfdpw,
+            RequestType::GETFDGR => self.ignore_getfdgr,
+            RequestType::GETFDHST => self.ignore_getfdhst,
+            RequestType::GETAI => self.ignore_getai,
+            RequestType::INITGROUPS => self.ignore_initgroups,
+            RequestType::GETSERVBYNAME => self.ignore_getservbyname,
+            RequestType::GETSERVBYPORT => self.ignore_getservbyport,
+            RequestType::GETFDSERV => self.ignore_getfdserv,
+            RequestType::GETNETGRENT => self.ignore_getnetgrent,
+            RequestType::INNETGR => self.ignore_innetgr,
+            RequestType::GETFDNETGR => self.ignore_getfdnetgr,
+            // This can't happen
+            RequestType::LASTREQ => false,
+        }
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            worker_count: 8,
+            handoff_timeout: Duration::from_secs(3),
+            ignore_getpwbyname: Default::default(),
+            ignore_getpwbyuid: Default::default(),
+            ignore_getgrbyname: Default::default(),
+            ignore_getgrbygid: Default::default(),
+            ignore_initgroups: Default::default(),
+            ignore_gethostbyaddr: Default::default(),
+            ignore_gethostbyaddrv6: Default::default(),
+            ignore_gethostbyname: Default::default(),
+            ignore_gethostbynamev6: Default::default(),
+            ignore_shutdown: Default::default(),
+            ignore_getstat: Default::default(),
+            ignore_invalidate: Default::default(),
+            ignore_getfdpw: Default::default(),
+            ignore_getfdgr: Default::default(),
+            ignore_getfdhst: Default::default(),
+            ignore_getai: Default::default(),
+            ignore_getservbyname: Default::default(),
+            ignore_getservbyport: Default::default(),
+            ignore_getfdserv: Default::default(),
+            ignore_getfdnetgr: Default::default(),
+            ignore_getnetgrent: Default::default(),
+            ignore_innetgr: Default::default(),
+        }
+    }
+}
+
+fn env_positive_usize(var: &'static str, default: usize) -> Result<usize> {
```

Comment:
nit: I generally prefer to flatten out something like this instead of an `if` nested inside an `if-let`. if it weren't for MSRV i'd suggest `guard-let`.

```rust
let s = match env::var(var) {
  Ok(s) => s,
  Err(_) => return default,
};

let int_value: usize = s.parse().with_context(|| "good error message")?;
if val > 0 {
  Ok(val)
} else {
   Err("better error message")
}
```

I would also be down for something like this, but having the default be a string always feels a little silly.

```rust
let int_val: usize = env::var(var).unwrap_or(default).parse().with_context(...)?;
if val > 0 {
  Ok(val)
} else {
   Err("better error message")
}
```